File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        
        int cnt = 0;
        int delZero = 0;
        int c = 0;
        
        while(!s.equals("1")) {
            String str = "";
            for(int i = 0; i < s.length(); i++) {
                if(s.charAt(i) == '0') {
                    delZero++;
                }
                if(s.charAt(i) == '1') {
                    str += s.charAt(i);
                }
            }
        c = str.length();
        String bin = Integer.toBinaryString(c);
        s = bin;
        cnt++;
        }
        int[] answer = {cnt, delZero};
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    var answer = [];
    let zeroCount = 0;
    let count = 0;
    while(true) {
        if(s === '1') break;
        count++;
        for(let i = 0; i < s.length; i++) {
            if (s[i] === '0') zeroCount++;
        }
        s = s.replaceAll('0','');
        s = s.length.toString(2);

    }
    
    answer = [count,zeroCount];
    return answer;
}